version.py

```diff
@@ -6,15 +6,15 @@
 
 from datetime import datetime,timezone
 from shutil import which
-from subprocess import check_output
+from subprocess import check_output, PIPE
 
 srcdir = os.path.dirname(os.path.abspath(sys.argv[0]))
 git_dir = os.path.join(srcdir, ".git")
 git = which('git')
 
 if git and os.path.exists(git_dir):
     version = check_output([git, "-C", srcdir, "describe", "--always", "--tags",
-                            "--dirty"], encoding="UTF-8")
+                            "--dirty"], stdin=PIPE, encoding="UTF-8")
```

Comment:
what's this for? we're not piping anything to it

Reply:
We don't, but waf does. It does this prefork magic and we have to have stdin available else it blow up. See https://github.com/mpv-player/mpv-build/issues/204

Reply:
I don't see how waf can be involved here at all, this is git being called from a python script that is ran (not imported!) from waf.
Are you saying `version.py` works if ran on its own in msys2 but not from waf?

Reply:
>  Are you saying version.py works if ran on its own in msys2 but not from waf?
Yes, exactly. waf is "preforking" processes and dispatch work to them, this has some implications how subprocesses are executed. And there are difference between python on linux and windows in regards to descriptors, obviously... I don't care about waf, I spend already too much time debugging this crap to understand why simple subprocess is failing when run within it. And here is the fix, which has no downsides and improve compatibility greatly. Feel free to suggest better solution.

Reply:
Sure that's okay, I was just wondering whether there wasn't a misunderstanding.
Can you add a comment that `stdin=PIPE` is only needed to work around waf bugs?

Reply:
I've added pretty vague comment about this. Frankly waf was annoying to work with, glad we have good meson buildsystem...